packages/js/uri-resolvers/src/helpers/RetryResolver.ts

```diff
@@ -0,0 +1,113 @@
+import { UriResolverLike } from "./UriResolverLike";
+import { UriResolver } from "./UriResolver";
+import { UriResolutionResult } from "./UriResolutionResult";
+
+import {
+  CoreClient,
+  IUriResolutionContext,
+  IUriResolver,
+  Result,
+  Uri,
+  UriPackageOrWrapper,
+} from "@polywrap/core-js";
+
+/** configure how additional resolution attempts are handled after an initial resolution attempt fails */
+export type RetryOptions = {
+  /** the number of additional resolution attempts */
+  retries: number;
+  /** the duration (in ms) to pause between attempts */
+  interval: number;
+};
+
+/** A map or uri or authority strings to retry options */
+export type RetryResolverOptions = {
+  [uriOrAuthority: string]: RetryOptions;
+};
+
+export class RetryResolver<TError = undefined> implements IUriResolver<TError> {
+  constructor(
+    private resolver: IUriResolver<TError>,
+    private options: RetryResolverOptions
```

Comment:
It'd be nice to not have to rely on authority for all cases here.
Like having a handler to decide if the request should retry or not.
Maybe both (e.g. a union type) ?

Reply:
Good idea! it could accept a callback